packages/react/src/menu/popup/MenuPopup.tsx

```diff
@@ -98,7 +98,8 @@ const MenuPopup = React.forwardRef(function MenuPopup(
       context={floatingContext}
       modal={false}
       disabled={!mounted}
-      initialFocus={nested ? -1 : 0}
+      returnFocus={parent.type === undefined}
+      initialFocus={parent.type === 'menu' ? -1 : 0}
```

Comment:
Playing with this, it doesn't seem to make sense not to return focus in a web context. Radix and Ariakit return focus to the trigger. In the OS, menubar has different accessibility semantics most likely regarding where focus is placed, but inside a browser the focus should be restored, otherwise it just restarts at the beginning of the document and loses the context